src/OrbitBase/include/OrbitBase/AnyErrorOf.h

```diff
@@ -0,0 +1,132 @@
+//    considered equal if `AnyErrorOf` holds a value of the comparing error type and if the equality
+//    comparison operator returns true.
+template <typename... ErrorTypes>
+class AnyErrorOf : public std::variant<ErrorTypes...> {
+  using Base = std::variant<ErrorTypes...>;
+
+ public:
+  using Base::Base;
+  using Base::operator=;
+
+  static_assert(ParameterPackTrait<AnyErrorOf, ErrorTypes...>::kSize >= 1,
+                "AnyError<> (AnyErrorOf with no error types) is not allowed.");
+
+  static_assert(!ParameterPackTrait<AnyErrorOf, ErrorTypes...>::kHasDuplicates,
+                "AnyError<ErrorTypes...> must not have duplicate error types.");
+
+  template <typename... Types>
+  constexpr static bool CanBeConstructedFromTypesAndIsNotCopy() {
+    constexpr ParameterPackTrait<AnyErrorOf, ErrorTypes...> kThisTrait{};
+    constexpr ParameterPackTrait<AnyErrorOf, Types...> kOtherTrait{};
+
+    // We return false if `AnyErrorOf<ErrorTypes...>` is the same type as `AnyErrorOf<Types...>`.
+    // This avoids collisions with the copy and move constructors/assignment operators.
+    if (kThisTrait == kOtherTrait) return false;
+
+    // We only allow conversion from an instance of `AnyErrorOf<Types...>` with `Types` being a
+    // subset of `ErrorTypes`.
+    return kThisTrait.IsSubset(kOtherTrait);
+  }
+
+  // Is true iff `Type` is in the `ErrorTypes...` parameter pack.
+  template <typename Type>
+  constexpr static bool kIsAnErrorType =
+      ParameterPackTrait<AnyErrorOf, ErrorTypes...>::template kContains<Type>;
+
+  // The following converting constructors/assignment operators allow conversion of any AnyErrorOf
+  // type into a compatible AnyErrorOf type. An AnyErrorOf type is considered compatible
+  // if its error type list is a super set of the other's error type list. The order of the types
+  // doesn't matter though.
+  //
+  // Examples:
+  //  - AnyErrorOf<E1> can be converted into AnyErrorOf<E1, E2> but not into AnyErrorOf<E2, E3>.
+  //  - AnyErrorOf<E1, E2> can be converted into AnyErrorOf<E1, E3, E2>.
+  //  - AnyErrorOf<E1, E2> can be converted into AnyErrorOf<E2, E1>.
+  template <typename... Types,
+            std::enable_if_t<CanBeConstructedFromTypesAndIsNotCopy<Types...>(), int> = 0>
+  // NOLINTNEXTLINE(google-explicit-constructor)
+  AnyErrorOf(const AnyErrorOf<Types...>& other)
+      : Base{std::visit([](const auto& alternative) -> Base { return alternative; }, other)} {}
+
+  template <typename... Types,
+            std::enable_if_t<CanBeConstructedFromTypesAndIsNotCopy<Types...>(), int> = 0>
+  // NOLINTNEXTLINE(google-explicit-constructor)
+  AnyErrorOf(AnyErrorOf<Types...>&& other)
+      : Base{std::visit([](auto&& alternative)
+                            -> Base { return std::forward<decltype(alternative)>(alternative); },
+                        std::move(other))} {}
+
+  template <typename... Types,
+            std::enable_if_t<CanBeConstructedFromTypesAndIsNotCopy<Types...>(), int> = 0>
```

Comment:
What would be the advantage of that? It looks less readable to me.

Reply:
```
std::enable_if_t<CanBeConstructedFromTypesAndIsNotCopy<Types...>(), int> = 0
```
vs 
```
EnableIfIntCanBeConstructedFromTypesAndIsNotCopy<Types...> = 0
```
The former is nested, so IMO it _is_ harder to read. Also, when I see code duplication, I always have the same voice in my head
1. Wow, looks similar
2. What's the difference? I must be missing smth
3. What? No difference??? Why have I just wasted time looking for a black cat in a dark room which didn't even had a can in it? 

OTOH, the latter might be hard to read because the name is ugly. Yet, it perfectly describes what the thing does.

Reply:
I tried it. I think readability is not suffering too much. Change comes in sec. Thanks.

Reply:
I like it much more now. Thanks. What's about `std::enable_if_t<B, int>`? It's OK if you consciously wish to keep it and it's not an oversight.

Reply:
I don't understand